checks/apps/quantumespresso/quantumespresso_check.py

```diff
@@ -48,106 +91,44 @@ def time(self):
-                    'MPICH_OFI_STARTUP_CONNECT': 1,
-                    'OMP_NUM_THREADS': 8,
-                    'OMP_PLACES': 'cores',
-                    'OMP_PROC_BIND': 'close'
-                }
-
-        else:
-            self.energy_reference = -11427.09017152
-            if self.current_system.name in ['daint']:
-                self.num_tasks = 576
-                self.num_tasks_per_node = 36
-            elif self.current_system.name in ['eiger', 'pilatus']:
-                self.num_tasks = 256
-                self.num_tasks_per_node = 16
-                self.num_cpus_per_task = 16
-                self.num_tasks_per_core = 1
-                self.env_vars = {
-                    'MPICH_OFI_STARTUP_CONNECT': 1,
-                    'OMP_NUM_THREADS': 8,
-                    'OMP_PLACES': 'cores',
-                    'OMP_PROC_BIND': 'close'
-                }
-
     @run_before('performance')
-    def set_reference(self):
-        references = {
-            'small': {
-                'dom:mc': {'time': (109.814, None, 0.10, 's')},
-                'daint:mc': {'time': (98.475, None, 0.10, 's')},
-                'eiger:mc': {'time': (55.51, None, 0.10, 's')},
-                'pilatus:mc': {'time': (50.981, None, 0.10, 's')}
-            },
-            'large': {
-                'daint:mc': {'time': (125.648, None, 0.10, 's')},
-                'eiger:mc': {'time': (47.165, None, 0.10, 's')},
-                'pilatus:mc': {'time': (43.211, None, 0.10, 's')}
-            }
+    def set_performance_reference(self):
+        self.reference = {
+            '*': {'time': (self.ref_dict['performance_reference'][0]['P'],
+                           None, 0.10, 's')}
         }
-        self.reference = references[self.scale]
-
-    @run_before('run')
-    def set_task_distribution(self):
-        self.job.options = ['--distribution=block:block']
-
-    @run_before('run')
-    def set_cpu_binding(self):
-        self.job.launcher.options = ['--cpu-bind=cores']
 
 
 @rfm.simple_test
-class QuantumESPRESSOGpuCheck(QuantumESPRESSOCheck):
-    valid_systems = ['daint:gpu']
-    num_gpus_per_node = 1
-    num_tasks_per_node = 1
-    num_cpus_per_task = 12
-    energy_tolerance = 1.0e-7
+class UENV_QuantumESPRESSOCheck(QuantumESPRESSOBase,
+                                ExtraLauncherOptionsMixin,
+                                CudaVisibleDevicesAllMixin):
+    valid_systems = ['+uenv -amdgpu']
+    valid_prog_environs = ['+quantum-espresso', '+mpi', '+openmp']
+    tags = {'production', 'uenv'}
+    energy_tolerance = 1.0e-6
+    test_name = parameter(['Au-surf'])
 
-    @run_after('init')
-    def setup_test(self):
-        self.descr = (f'QuantumESPRESSO GPU check (version: {self.scale}, '
-                      f'{self.variant})')
-        if self.scale == 'small':
-            self.valid_systems += ['dom:gpu']
-            self.num_tasks = 6
-            self.energy_reference = -11427.09017168
+@rfm.simple_test
```

Comment:
```suggestion

@rfm.simple_test
```

Reply:
again, I don't understand this one. two spaces between tests?